api_spec.yaml

```diff
@@ -1377,7 +1378,140 @@ paths:
+    get:
+      tags:
+        - User Metadata Recipe
+      operationId: userMetadataRead
+      description: |
+        Gets the stored metadata object of the user
+      parameters:
+        - $ref: '#/components/parameters/userMetadataRid'
+        - $ref: '#/components/parameters/api-key'
+        - $ref: '#/components/parameters/cdi-version'
+        - $ref: '#/components/parameters/userId'
+      responses:
+        '200':
+          description: 'The operation was successful'
+          content:
+            application/json:
+                schema: 
+                  type: object
+                  properties:
+                    status:
+                      $ref: '#/components/schemas/statusOK'
+                    metadata:
+                      $ref: '#/components/schemas/userMetadata'
+        '400':
+          $ref: '#/components/responses/400'
+        
+        '401':
+          $ref: '#/components/responses/401'
+        
+        '404':
+          $ref: '#/components/responses/404'
+        
+        '500':
+          $ref: '#/components/responses/500'
+
+    post:
+      tags:
+        - User Metadata Recipe
+      operationId: userMetadataUpdate
+      description: |
+        Updates the metadata object stored about the user by doing a shallow merge of the stored and the update JSONs
+        and removing properties set to null on the root level of the update object. The merged object is then reserialized and stored.
+
+        e.g.:
+          - stored: `{ "preferences": { "theme":"dark" }, "notifications": { "email": true }, "todos": ["example"] }`
+
+          - update: `{ "notifications": { "sms": true }, "todos": null }`
+
+          - result: `{ "preferences": { "theme":"dark" }, "notifications": { "sms": true } }`
+
+      parameters:
+        - $ref: '#/components/parameters/userMetadataRid'
+        - $ref: '#/components/parameters/api-key'
+        - $ref: '#/components/parameters/cdi-version'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              properties:
+                userId:
+                  $ref: '#/components/schemas/userId'
+                metadataUpdate:
+                  $ref: '#/components/schemas/userMetadataUpdate'
+      responses:
+        '200':
+          description: 'The operation was successful'
+          content:
+            application/json:
+                schema: 
+                  type: object
+                  properties:
+                    status:
+                      $ref: '#/components/schemas/statusOK'
+                    metadata:
+                      $ref: '#/components/schemas/userMetadata'
+
+        '400':
+          $ref: '#/components/responses/400'
+        
+        '401':
+          $ref: '#/components/responses/401'
+        
+        '404':
+          $ref: '#/components/responses/404'
+        
+        '500':
+          $ref: '#/components/responses/500'
+
+  /recipe/userMetadata/delete:
```

Comment:
We use `/remove` instead of `/delete` everywhere.

Reply:
Is this API needed? Cause it's equal to just updating a user's metadata to an empty object?

Reply:
If we go with the merging update mechanism it's simpler and more convenient, since the empty object update wouldn't change anything, you'd have to explicitly null out everything.

Reply:
Alright! Then we can keep this in case we are going with the merge + update route. Else we can remove this.

Reply:
Since we are going with shallow merge, we should keep this API.